migrations/20231129172339_job_queue_table.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TYPE job_status AS ENUM(
+    'available',
+    'completed',
+    'failed',
+    'running'
+);
+
+CREATE TABLE job_queue(
+    id BIGSERIAL PRIMARY KEY,
+    attempt INT NOT NULL DEFAULT 0,
+    attempted_at TIMESTAMPTZ DEFAULT NULL,
+    attempted_by TEXT[] DEFAULT ARRAY[]::TEXT[],
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    finished_at TIMESTAMPTZ DEFAULT NULL,
+    started_at TIMESTAMPTZ DEFAULT NULL,
```

Comment:
how are we tracking how long we waited for the endpoint - e.g. if someone is close to the endpoint being too slow we could warn them in the UI and also useful info for us in case of lag in processing 🤷‍♀️

Reply:
`started_at` (now renamed to `attempted_at`) and `finished_at` capture whatever time it took the worker to do the work. If the work is making a fetch call, then that's what we are recording, but the queue is work-agnostic.